src/lib.rs

```diff
@@ -256,7 +254,10 @@ impl MyDrone {
                 .collect();
 
             for next_hop in &next_hops_to_forward_packet_to {
-                let routing_header = previous_routing_header.clone();
+                let routing_header = SourceRoutingHeader {
+                    hop_index: 0,
+                    hops: vec![],
+                };
```

Comment:
Here it's ok because flood request ignores header, given that this is a behavior not clearly defined in the protocol maybe we should add it to the readme @saussice9 if you want to write something about the fact that when our drone forwards flood requests it resets the sourcerouting header to an empty one